From bdaa313d14c8069ae441d417b2e9be5cce244bca Mon Sep 17 00:00:00 2001
From: Daniel Scally <djrscally@gmail.com>
Date: Sat, 24 Oct 2020 22:37:33 +0100
Subject: [PATCH 01/13] property: Add support to
 fwnode_graph_get_endpoint_by_id() for parsing fwnode->secondary if no
 endpoint is found under the primary

fwnode_handles created from software_nodes can be used to construct
media graphs for devices where the firmware is lacking. Implementation
of this method requires fwnode_graph_get_endpoint_by_id() to also check
fwnode->secondary in the event that it fails to find a valid endpoint
against the primary.

Signed-off-by: Dan Scally <djrscally@gmail.com>
---
 drivers/base/property.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/base/property.c b/drivers/base/property.c
index 4c43d30145c6..c7f8dab6967d 100644
--- a/drivers/base/property.c
+++ b/drivers/base/property.c
@@ -1157,6 +1157,10 @@ fwnode_graph_get_endpoint_by_id(const struct fwnode_handle *fwnode,
 		best_ep_id = fwnode_ep.id;
 	}
 
+	if (!best_ep && !IS_ERR_OR_NULL(fwnode->secondary))
+		best_ep = fwnode_graph_get_endpoint_by_id(fwnode->secondary,
+							  port, endpoint, flags);
+
 	return best_ep;
 }
 EXPORT_SYMBOL_GPL(fwnode_graph_get_endpoint_by_id);
-- 
2.25.1

